src/plugins/intel_cpu/src/nodes/executors/x64/jit_matmul_small.cpp

```diff
@@ -0,0 +1,198 @@
+                                                                                                      post_op,
+                                                                                                      vmm_d_weights,
+                                                                                                      vmm_d_bias,
+                                                                                                      reg_d_weights,
+                                                                                                      reg_d_bias));
+        }
+    }
+
+    this->preamble();
+
+    mov(reg_input1, ptr[reg_params + GET_OFF(input1)]);
+    mov(reg_input2, ptr[reg_params + GET_OFF(input2)]);
+    mov(reg_out, ptr[reg_params + GET_OFF(output)]);
+    mov(reg_work_amount, ptr[reg_params + GET_OFF(B)]);
+    if (jcp_.M > 2 || jcp_.N > 2 || jcp_.K > 2) {
+        assert("matmul_small_kernel only support M/N/K smaller than 3.");
+    }
+
+    if (attr_.post_ops_.len() != 0) {
+        mov(reg_post_ops_data, ptr[reg_params + GET_OFF(post_op_data)]);
+        mov(reg_oc_off, ptr[reg_params + GET_OFF(oc_off)]);
+    }
+
+    Xbyak::Label loop_label;
+    Xbyak::Label loop_end_label;
+    Xbyak::Label offset_reset_label;
+    L(loop_label);
+    {
+        cmp(reg_work_amount, 1);
+        jl(loop_end_label, T_NEAR);
+
+        // loop unrolling and register utilization in each batch
+        // load
+        for (size_t m = 0; m < jcp_.M; m++) {
+            for (size_t k = 0; k < jcp_.K; k++) {
+                uni_vmovss(vmm_input1[m * jcp_.K + k], ptr[reg_input1]);
+                add(reg_input1, sizeof(float));
+            }
+        }
+        for (size_t k = 0; k < jcp_.K; k++) {
+            for (size_t n = 0; n < jcp_.N; n++) {
+                uni_vmovss(vmm_input2[k * jcp_.N + n], ptr[reg_input2]);
+                add(reg_input2, sizeof(float));
+            }
+        }
+
+        for (size_t m = 0; m < jcp_.M; m++) {
+            for (size_t n = 0; n < jcp_.N; n++) {
+                uni_vpxor(vmm_output[m * jcp_.N + n], vmm_output[m * jcp_.N + n], vmm_output[m * jcp_.N + n]);
+            }
+        }
+        // outer most K to reduce RAW dependency.
+        for (size_t k = 0; k < jcp_.K; k++) {
+            for (size_t m = 0; m < jcp_.M; m++) {
+                for (size_t n = 0; n < jcp_.N; n++) {
+                    uni_vfmadd231ps(vmm_output[m * jcp_.N + n], vmm_input1[m * jcp_.K + k], vmm_input2[k * jcp_.N + n]);
+                }
+            }
+        }
+
+        // store
+        for (size_t m = 0; m < jcp_.M; m++) {
+            for (size_t n = 0; n < jcp_.N; n++) {
+                if (attr_.post_ops_.len() != 0) {
+                    apply_post_ops(element::f32, vmm_output[m * jcp_.N + n].getIdx(), true);
+                }
+                uni_vmovss(ptr[reg_out], vmm_output[m * jcp_.N + n]);
+                add(reg_out, sizeof(float));
```

Comment:
here only scalar form of SIMD instruction is used, maybe we can further increase the performance by packing in1/in2 from different batch.

Reply:
Regarding repack to enable SIMD fma, I actually discussed it with @maxnick several days ago. For small matmul, repacked src and weight are not reused as much as big matmul. Reapck itself consume too much time compared with fma. The algorithm complexity could even higher for repack version(C*M*K*N/SIMD_width + C*(M*K+K*N) > C*M*K*N  for M=2, K=2, N=1 in sam model case on avx2). Thanks Tingqian!